Fails to build C extension on i386

Building of C extension for zope.proxy-4.6.1 on i386 with clang16 with python3.9 on FreeBSD fails with incompatible function pointer types error. Everything is fine on amd64 arch. Would you suggest solution for this one?

```c
running build_ext
building 'zope.proxy._zope_proxy_proxy' extension
cc -Wno-unused-result -Wsign-compare -Wunreachable-code -DNDEBUG -O2 -pipe -fstack-protector-strong -fno-strict-aliasing -O2 -pipe -fstack-protector-strong -fno-strict-aliasing -fPIC -I/usr/local/include/python3.9 -c src/zope/proxy/_zope_proxy_proxy.c -o build/temp.freebsd-14.0-BETA4-i386-cpython-39/src/zope/proxy/_zope_proxy_proxy.o
src/zope/proxy/_zope_proxy_proxy.c:381:16: warning: 'PyEval_CallObjectWithKeywords' is deprecated [-Wdeprecated-declarations]
        return PyEval_CallObjectWithKeywords(Proxy_GET_OBJECT(self),
               ^
/usr/local/include/python3.9/ceval.h:17:1: note: 'PyEval_CallObjectWithKeywords' has been explicitly marked deprecated here
Py_DEPRECATED(3.9) PyAPI_FUNC(PyObject *) PyEval_CallObjectWithKeywords(
^
/usr/local/include/python3.9/pyport.h:508:54: note: expanded from macro 'Py_DEPRECATED'
#define Py_DEPRECATED(VERSION_UNUSED) __attribute__((__deprecated__))
                                                     ^
src/zope/proxy/_zope_proxy_proxy.c:887:5: error: incompatible function pointer types initializing 'hashfunc' (aka 'int (*)(struct _object *)') with an expression of type 'long (PyObject *)' (aka 'long (struct _object *)') [-Wincompatible-function-pointer-types]
    wrap_hash, /* tp_hash */
    ^~~~~~~~~
```

--- a/CHANGES.rst
+++ b/CHANGES.rst
@@ -7,6 +7,9 @@
 
 - Build Windows wheels on GHA.
 
+- Correct return type for ``wrap_hash`` (fixes build on ``i386``).
+  (`#61 <https://github.com/zopefoundation/zope.proxy/issues/61>`_)
+
 
 5.2 (2024-02-09)
 ================

--- a/src/zope/proxy/_zope_proxy_proxy.c
+++ b/src/zope/proxy/_zope_proxy_proxy.c
@@ -311,7 +311,7 @@ wrap_repr(PyObject *wrapper)
     return PyObject_Repr(Proxy_GET_OBJECT(wrapper));
 }
 
-static long
+static Py_hash_t
 wrap_hash(PyObject *self)
 {
     return PyObject_Hash(Proxy_GET_OBJECT(self));

